File: ICPC/Nacional 2025/D.py
import sys

binom = []

def calcBinom():
    for i in range(0, 61):
        binom.append(list())
        for j in range(0, 61):
            binom[i].append(0)
    for i in range(0, 61):
        binom[i][0] = 1
        binom[i][i] = 1

    for i in range(1, 61):
        for j in range(1, 61):
            binom[i][j] = binom[i-1][j-1] + binom[i-1][j]

def main():
    while True:
        try:
            read = input()
            numbers = read.split()
            numbers = list(map(int, numbers))
            n = numbers[0]
            k = numbers[1]

            if k == 0 and n == 0:
                return

            l = 0
            r = binom[n][k]

            read = input()
            perm = read.split()
            perm = list(map(int, perm))
            last = 1

            for i in range(0, k):
                for j in range(last, n+1):
                    # print(f"i {i}, j {j}")
                    if perm[i] == j:
                        r = l + binom[n - j][k - i - 1]
                        last = j + 1
                        break
                    else:
                        l += binom[n - j][k - i - 1]

            print(l)
        
        except EOFError:
            break

calcBinom()
main()

File: ICPC/Nacional 2025/D.cpp
#include<bits/stdc++.h>
using namespace std;
#define endl '\n'
#define pb push_back
#define sz(a) ((int)a.size())
#define all(a) a.begin(), a.end()
#define fi first
#define se second
#define lb lower_bound
#define ub upper_bound
typedef long long ll;
typedef long double ld;
typedef pair<int, int> pii;
// #include<ext/pb_ds/assoc_container.hpp>
// using namespace __gnu_pbds;
// using indexed_set = tree<int, null_type, less<int>, rb_tree_tag, tree_order_statistics_node_update>;

ll binom[61][61];
void calcBinom(){
    for(int i = 0; i < 61; i++) binom[i][i] = binom[i][0] = 1LL;
    for(int i = 1; i < 61; i++){
        for(int j = 1; j < i; j++){
            binom[i][j] = binom[i-1][j-1] + binom[i-1][j];
        }
    }
}

void solver(int n, int k){
    vector<int> perm(n);
    for(int i = 0; i < k; i++) cin>>perm[i];

    ll l = 0;
    int last = 1;
    for(int i = 0; i < k; i++){
        for(int j = last; j <= n; j++){
            if(j == perm[i]){
                last = j + 1;
                break;
            }
            l += binom[n - j][k - i - 1];
        }
    }

    cout<<l<<endl;
}

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);cout.tie(0);
    // freopen("name.in", "r", stdin);
	// freopen("name.out", "w", stdout);
    int t = 1;
    // cin>>t;
    calcBinom();
    int n, k;
    while(cin>>n>>k){
        solver(n, k);
    }

    return 0;
}